lib/cog/models/chat_handle.ex

```diff
@@ -19,8 +19,16 @@ defmodule Cog.Models.ChatHandle do
   detail_fields [:id, :handle, :user, :chat_provider, :chat_provider_user_id]
 
   def changeset(model, params \\ :empty) do
+    params = coerce_chat_provider_user_id(params)
+
     model
     |> cast(params, @required_fields, @optional_fields)
   end
 
+  defp coerce_chat_provider_user_id(%{"chat_provider_user_id" => chat_provider_user_id} = params),
```

Comment:
Me to. But, we never get past the `cast` which creates the changeset that we can add values to, because the integer and string types don't match up. I figured `cast` would try to coerce the value, but it doesn't. So we have to coerce on our own first and then push it into the `cast` call. The other option would be to use a virtual field with an integer type, but we also have `id`s that are strings so that wouldn't work well either. Really the cast call should allow you pass in functions to use for type conversions.